src/vs/editor/browser/widget/diffEditorWidget.ts

```diff
@@ -929,11 +929,14 @@ export class DiffEditorWidget extends Disposable implements editorBrowser.IDiffE
 
 		let diffDecorations = this._strategy.getEditorsDiffDecorations(lineChanges, this._ignoreTrimWhitespace, this._renderIndicators, foreignOriginal, foreignModified, this.originalEditor, this.modifiedEditor);
 
+		const prevState = this.modifiedEditor.saveViewState();
+
```

Comment:
The only view state that changes while applying these decorations is the scroll state. The scroll state is actually restored in the modified editor. See the last boolean argument passed into `this._modifiedEditorState.apply`, in the `apply` method that is the `restoreScrollState` argument.